Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -0,0 +1,1122 @@
+`f` descents to a map `LocalizedModule M S → α`.
+-/
+def liftOn {α : Type _} (x : LocalizedModule S M) (f : M × S → α)
+    (wd : ∀ (p p' : M × S), p ≈ p' → f p = f p') : α :=
+  Quotient.liftOn x f wd
+#align localized_module.lift_on LocalizedModule.liftOn
+
+theorem liftOn_mk {α : Type _} {f : M × S → α} (wd : ∀ (p p' : M × S), p ≈ p' → f p = f p')
+    (m : M) (s : S) : liftOn (mk m s) f wd = f ⟨m, s⟩ := by convert Quotient.liftOn_mk f wd ⟨m, s⟩
+#align localized_module.lift_on_mk LocalizedModule.liftOn_mk
+
+/-- If `f : M × S → M × S → α` respects the equivalence relation `LocalizedModule.r`, then
+`f` descents to a map `LocalizedModule M S → LocalizedModule M S → α`.
+-/
+def liftOn₂ {α : Type _} (x y : LocalizedModule S M) (f : M × S → M × S → α)
+    (wd : ∀ (p q p' q' : M × S), p ≈ p' → q ≈ q' → f p q = f p' q') : α :=
+  Quotient.liftOn₂ x y f wd
+#align localized_module.lift_on₂ LocalizedModule.liftOn₂
+
+theorem liftOn₂_mk {α : Type _} (f : M × S → M × S → α)
+    (wd : ∀ (p q p' q' : M × S), p ≈ p' → q ≈ q' → f p q = f p' q') (m m' : M)
+    (s s' : S) : liftOn₂ (mk m s) (mk m' s') f wd = f ⟨m, s⟩ ⟨m', s'⟩ := by
+  convert Quotient.liftOn₂_mk f wd _ _
+#align localized_module.lift_on₂_mk LocalizedModule.liftOn₂_mk
+
+instance : Zero (LocalizedModule S M) :=
+  ⟨mk 0 1⟩
+
+@[simp]
+theorem zero_mk (s : S) : mk (0 : M) s = 0 :=
+  mk_eq.mpr ⟨1, by rw [one_smul, smul_zero, smul_zero, one_smul]⟩
+#align localized_module.zero_mk LocalizedModule.zero_mk
+
+instance : Add (LocalizedModule S M)
+    where add p1 p2 :=
+    liftOn₂ p1 p2 (fun x y => mk (y.2 • x.1 + x.2 • y.1) (x.2 * y.2)) <|
+      fun ⟨m1, s1⟩ ⟨m2, s2⟩ ⟨m1', s1'⟩ ⟨m2', s2'⟩ ⟨u1, hu1⟩ ⟨u2, hu2⟩ =>
+          mk_eq.mpr
+            ⟨u1 * u2, by
+              -- Put everything in the same shape, sorting the terms using `simp`
+              have hu1' := congr_arg ((· • ·) (u2 * s2 * s2')) hu1
+              have hu2' := congr_arg ((· • ·) (u1 * s1 * s1')) hu2
+              simp only [smul_add, ← mul_smul, smul_assoc, mul_assoc, mul_comm,
+                mul_left_comm] at hu1' hu2' ⊢
+              rw [hu1', hu2']⟩
+
+theorem mk_add_mk {m1 m2 : M} {s1 s2 : S} :
+    mk m1 s1 + mk m2 s2 = mk (s2 • m1 + s1 • m2) (s1 * s2) :=
+-- Porting note: Simplified thanks to [linter.unreachableTactic].
+  mk_eq.mpr <| ⟨1, rfl⟩
+#align localized_module.mk_add_mk LocalizedModule.mk_add_mk
+
+private theorem add_assoc' (x y z : LocalizedModule S M) : x + y + z = x + (y + z) := by
+  induction' x using LocalizedModule.induction_on with mx sx
+  induction' y using LocalizedModule.induction_on with my sy
+  induction' z using LocalizedModule.induction_on with mz sz
+  simp only [mk_add_mk, smul_add]
+  refine' mk_eq.mpr ⟨1, _⟩
+  rw [one_smul, one_smul]
+  congr 1
+  · rw [mul_assoc]
+  · rw [eq_comm, mul_comm, add_assoc, mul_smul, mul_smul, ← mul_smul sx sz, mul_comm, mul_smul]
+-- #align localized_module.add_assoc' localized_module.add_assoc'
```

Comment:
Why are these `#align`'s commented out? Perhaps leave a comment explaining.

(If the reason is that you made something `private`, either don't make it private (which is usually a bad idea) or explain why.)